src/commands/dsyms/utils.ts

```diff
@@ -26,64 +17,37 @@ export const isZipFile = async (filepath: string) => {
   }
 }
 
-export const getMatchingDSYMFiles = async (
-  absoluteFolderPath: string,
-  context: BaseContext
-): Promise<(Dsym | undefined)[]> => {
-  const dSYMFiles = await globAsync(buildPath(absoluteFolderPath, '**/*.dSYM'), {})
-
-  const allDsyms = dSYMFiles.map(async (dSYMPath) => {
-    try {
-      const uuids = await dwarfdumpUUID(dSYMPath)
-
-      return new Dsym(dSYMPath, uuids)
-    } catch {
-      context.stdout.write(renderInvalidDsymWarning(dSYMPath))
-
-      return undefined
-    }
-  })
+export const createUniqueTmpDirectory = async (): Promise<string> => {
+  const uniqueValue = Math.random() * Number.MAX_SAFE_INTEGER
+  const directoryPath = buildPath(tmpdir(), uniqueValue.toString())
+  await promises.mkdir(directoryPath, {recursive: true})
 
-  return Promise.all(allDsyms)
+  return directoryPath
 }
 
-export const dwarfdumpUUID = async (filePath: string) => {
-  const output = await execute(`dwarfdump --uuid '${filePath}'`)
-
-  const uuids: string[] = []
-  output.stdout.split('\n').forEach((line: string) => {
-    const regexMatches = line.match(UUID_REGEX)
-    if (regexMatches && regexMatches.length > 0) {
-      uuids.push(regexMatches[0])
-    }
-  })
-
-  return uuids
+export const deleteDirectory = async (directoryPath: string) => {
+  await execute(`rm -rf '${directoryPath}'`)
 }
```

Comment:
@lraucy @m-rousse I'm not OK with this `rm -rf` as it might be dangerous. Unfortunately, I couldn't find other way for deleting `<tmp>/datadog-ci` recursively with our `node` setup. [Here](https://github.com/DataDog/datadog-ci/pull/300#discussion_r683321374) we mentioned [rimraf](https://www.npmjs.com/package/rimraf) - do you think this is sth I could look into?

Alternatively, as I know the intermediate directories structure I could iterate it recursively and use non-recursive `fs.unlink()` on each file starting from leaf elements. Not sure if that would be generic solution though.

Otherwise, as we only create directories inside `/tmp` we could orphan it and let the OS flush it when needed. I'm not a fan of this neither - now when receiving 250MB of fat dSYMs at input, we will be creating 250MB of slimmed dSYMs in `<tmp>/.../intermediate/` and ~70MB of compressed payloads in `<tmp>/.../upload/`.

Reply:
I'd be in favor of introducing `rimraf` for this purpose :+1:

Reply:
Thanks @m-rousse. I did it (https://github.com/DataDog/datadog-ci/pull/500/commits/7b47146b0a34c8d780460a262b613331010b5df7). I had to add [`rimraf`](https://www.npmjs.com/package/rimraf) and [`@types/rimraf`](https://www.npmjs.com/package/@types/rimraf) for TS support. I hope that's correct (please confirm, I'm very new to `node` / `yarn` / `ts`).

Reply:
This looks good to me :+1: